fetchcode/giturl.py

```diff
@@ -0,0 +1,59 @@
+# fetchcode is a free software tool from nexB Inc. and others.
+# Visit https://github.com/nexB/fetchcode for support and download.
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and http://aboutcode.org
+#
+# This software is licensed under the Apache License version 2.0.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: 
+# http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import os
+
+from urllib.parse import urlparse, urlunparse
+
+
+def parse_url(url):
+	return urlparse(url)
+
+def canHandle(source):
+	url_parts = parse_url(source)
+	if url_parts.scheme not in ('http', 'https', 'git', ''):
+		return False
+	elif not url_parts.scheme:
+		return os.path.exists(os.path.join(source, '.git'))
+	else:
+		return True
+
+def clone(source, dest, branch='master', depth=None):
+	if not canHandle(source):
+		raise UnhandledSource("Cannot handle {}".format(source))
+	if os.path.exists(dest):
+		cmd = ['git', '-C', dest, 'pull', source, branch]
+	else:
+		cmd = ['git', 'clone', source, dest, '--branch', branch]
+		if depth:
+			cmd.extend(['--depth', depth])
+	os.system(' '.join(cmd))
+
+def install(source, dest, branch='master', depth=None):
+	url_parts = parse_url(source)
+	branch_name = url_parts.path.strip("/").split("/")[-1]
+	if dest:
+		dest_dir = os.path.join(dest, branch_name)
+	else:
+		dest_dir = os.path.join(os.environ.get('CHARM_DIR'), "fetched",
```

Comment:
Whats is CHARM_DIR?

Reply:
Since you copied this code from https://github.com/juju/charm-helpers/blob/master/charmhelpers/fetch/giturl.py or similar we absolutely need to track its origin and license AND keep all original notice.

**We never borrow or copy  code without tracking it and documenting where it comes from.**

Now, why not reusing charm-helpers as a library directly?

Reply:
I think instead of using their module I should code mine with some little tweaks and how can I track its origin, please help me in that